packages/gallery/src/components/gallery/leanGallery/leanGallery.js

```diff
@@ -94,18 +94,43 @@ export default class LeanGallery extends React.Component {
     const { styles } = this.props;
     const { gridStyle, numberOfImagesPerRow, imageMargin } = styles;
 
-    // const minmaxFix = 0.75; //this fix is meant to compensate for the css grid ability to use the number as a minimum only (the pro-gallery is trying to get as close as possible to this number)
-    const minmaxFix = 1;
+    const minmaxFix = 1; //this fix is meant to compensate for the css grid ability to use the number as a minimum only (the pro-gallery is trying to get as close as possible to this number)
     const itemSize = this.calcItemSize() * minmaxFix;
 
-    const gridTemplateColumns = gridStyle === 1 ? `repeat(${numberOfImagesPerRow}, 1fr)` : `repeat(auto-fit, minmax(${itemSize}px, 1fr))`;
+    const numberOfColumns = gridStyle === GALLERY_CONSTS.gridStyle.SET_ITEMS_PER_ROW ? numberOfImagesPerRow : this.calcNumberOfColumns();
+    const gridTemplateColumns = numberOfColumns > 0 ? `repeat(${numberOfColumns}, 1fr)` : `repeat(auto-fit, minmax(${itemSize}px, 1fr))`;
 
     return {
       gridTemplateColumns,
       gridGap: `${imageMargin}px`
     };
   }
 
+  calcNumberOfColumns() {
```

Comment:
this function should only happen after componentDidMount. Otherwise, the galleryWidth is meaningless.
Call it on componentDidMount and store the value in the state. later, in createGalleryStyles, use that state value (if greater than 0)